internal/app/orchestrator/orchestrator.go

```diff
@@ -0,0 +1,31 @@
+package orchestrator
+
+import (
+	gcp "github.com/envoyproxy/go-control-plane/pkg/cache"
+)
+
+// Orchestrator provides an interface that handles the discovery requests from sidecars.
+type Orchestrator interface {
+	// CreateWatch returns a new open watch from a non-empty request.
+	// Returns a channel with Response and a function which is a callback for handling stream cancellations.
+	CreateWatch(req gcp.Request) (chan gcp.Response, func())
```

Comment:
Do we need an error code param in the callback func?

Reply:
This is coming from https://github.com/envoyproxy/go-control-plane/blob/master/pkg/cache/v2/cache.go#L43 , i don;t think we can introduce params.